signature/envelope.go

```diff
@@ -0,0 +1,96 @@
+// Package signature provides operations for types that implement
+// signature.Envelope or signature.Signer.
+//
+// An Envelope is a structure that creates and verifies a signature using the
+// specified signing algorithm with required validation. To register a new
+// envelope, call RegisterEnvelopeType first during the initialization.
+//
+// A Signer is a structure used to sign payload generated after signature
+// envelope created. The underlying signing logic is provided by the underlying
+// local crypto library or the external signing plugin.
+package signature
+
+import "fmt"
+
+// Envelope provides functions to basic functions to manipulate signatures.
+type Envelope interface {
+	// Sign generates and sign the envelope according to the sign request.
+	Sign(req *SignRequest) ([]byte, error)
+
+	// Verify verifies the envelope and returns its enclosed payload and signer
+	// info.
+	Verify() (*Payload, *SignerInfo, error)
+
+	// Payload returns the payload of the envelope.
+	// Payload is trusted only after the successful call to `Verify()`.
+	Payload() (*Payload, error)
+
+	// SignerInfo returns the signer information of the envelope.
+	// SignerInfo is trusted only after the successful call to `Verify()`.
+	SignerInfo() (*SignerInfo, error)
```

Comment:
SignerInfo may not be best name, consider SignatureInfo or similar. Payload should be part of this struct.

Reply:
I was thinking in the same line for Payload() (*Payload, error) - Payload is any payload in general, so should be call it Payload Envelope().

Reply:
How about
```go
Content() (*EnvelopeContent, error)
```

Reply:
or
```
Data() (*EnvelopeData, error)
```

Reply:
where `EnvelopeConent` is a mimic of `SignedData` in CMS.

For CMS structs, see https://github.com/notaryproject/notation-core-go/blob/main/internal/crypto/cms/cms.go

Reply:
thanks for the advice. I feel `Content()` sounds better to me.

Reply:
I vote for content. Like it !! Thank You